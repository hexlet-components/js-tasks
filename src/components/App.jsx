import React from 'react'; // eslint-disable-line
import NewTaskFormContainer from '../containers/NewTaskForm';
import TasksListContainer from '../containers/TasksList';

export default () => (
  <div className="col-5">
    <NewTaskFormContainer />
    <TasksListContainer />
  </div>
);
