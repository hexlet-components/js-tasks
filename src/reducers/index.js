import _ from 'lodash';
import { combineReducers } from 'redux';
import { handleActions } from 'redux-actions';
import { reducer as formReducer } from 'redux-form';
import * as actions from '../actions';

const taskCreatingState = handleActions({
  [actions.addTaskRequest]() {
    return 'requested';
  },
  [actions.addTaskFailure]() {
    return 'failed';
  },
  [actions.addTaskSuccess]() {
    return 'successed';
  },
}, 'none');

const taskUpdatingState = handleActions({
  [actions.updateTaskRequest]() {
    return 'requested';
  },
  [actions.updateTaskFailure]() {
    return 'failed';
  },
  [actions.updateTaskSuccess]() {
    return 'successed';
  },
}, 'none');

const tasks = handleActions({
  [actions.addTaskSuccess](state, { payload: { task } }) {
    return { ...state, [task.id]: task };
  },
  [actions.removeTaskSuccess](state, { payload: { task } }) {
    return _.omit(state, task.id);
  },
  [actions.updateTaskSuccess](state, { payload: { task } }) {
    const oldTask = state[task.id];
    const updatedTask = { ...oldTask, ...task };
    return { ...state, [task.id]: updatedTask };
  },
  [actions.toggleTaskState](state, { payload: { id } }) {
    const task = state[id];
    const newState = task.state === 'active' ? 'finished' : 'active';
    const updatedTask = { ...task, state: newState };
    return { ...state, [task.id]: updatedTask };
  },
}, {});

export default combineReducers({
  form: formReducer,
  taskCreatingState,
  taskUpdatingState,
  tasks,
});
