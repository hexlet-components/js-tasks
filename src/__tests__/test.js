import 'js-polyfills/html';

import React from 'react' // eslint-disable-line
import { Provider } from 'react-redux' // eslint-disable-line
import { createStore } from 'redux' // eslint-disable-line
import Enzyme, { mount, render } from 'enzyme'; // eslint-disable-line
import Adapter from 'enzyme-adapter-react-16'; // eslint-disable-line
import reducers from '../reducers';
import App from '../components/App.jsx';

Enzyme.configure({ adapter: new Adapter() });

test('Store', () => {
  const store = createStore(reducers);

  const wrapper = mount(
    <Provider store={store}>
      <App />
    </Provider>,
    document.getElementById('container'),
  );
  expect(render(wrapper)).toMatchSnapshot();

  const newTaskInput = wrapper.find('input[type="text"]');
  const newTaskSubmit = wrapper.find('button[type="submit"]');

  newTaskInput.simulate('change', { target: { value: 'na-na' } });
  expect(render(wrapper)).toMatchSnapshot();

  newTaskSubmit.simulate('submit');
  expect(render(wrapper)).toMatchSnapshot();

  newTaskInput.simulate('change', { target: { value: 'another task' } });
  expect(render(wrapper)).toMatchSnapshot();

  newTaskSubmit.simulate('submit');
  expect(render(wrapper)).toMatchSnapshot();
});
