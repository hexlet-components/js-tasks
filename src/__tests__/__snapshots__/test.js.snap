// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store 1`] = `
<div
  class="col-5"
>
  <form
    action=""
    class="form-inline"
  >
    <div
      class="form-group mx-sm-3"
    >
      <input
        required=""
        type="text"
        value=""
      >
        
      </input>
    </div>
    <button
      class="btn btn-primary btn-sm"
      type="submit"
    >
      Add
    </button>
  </form>
</div>
`;

exports[`Store 2`] = `
<div
  class="col-5"
>
  <form
    action=""
    class="form-inline"
  >
    <div
      class="form-group mx-sm-3"
    >
      <input
        required=""
        type="text"
        value="na-na"
      >
        
      </input>
    </div>
    <button
      class="btn btn-primary btn-sm"
      type="submit"
    >
      Add
    </button>
  </form>
</div>
`;

exports[`Store 3`] = `
<div
  class="col-5"
>
  <form
    action=""
    class="form-inline"
  >
    <div
      class="form-group mx-sm-3"
    >
      <input
        required=""
        type="text"
        value=""
      >
        
      </input>
    </div>
    <button
      class="btn btn-primary btn-sm"
      type="submit"
    >
      Add
    </button>
  </form>
  <div
    class="row"
  >
    <div
      class="col-10"
    >
      na-na
    </div>
    <div
      class="col"
    >
      <a
        href="#"
      >
        x
      </a>
    </div>
  </div>
</div>
`;

exports[`Store 4`] = `
<div
  class="col-5"
>
  <form
    action=""
    class="form-inline"
  >
    <div
      class="form-group mx-sm-3"
    >
      <input
        required=""
        type="text"
        value="another task"
      >
        
      </input>
    </div>
    <button
      class="btn btn-primary btn-sm"
      type="submit"
    >
      Add
    </button>
  </form>
  <div
    class="row"
  >
    <div
      class="col-10"
    >
      na-na
    </div>
    <div
      class="col"
    >
      <a
        href="#"
      >
        x
      </a>
    </div>
  </div>
</div>
`;

exports[`Store 5`] = `
<div
  class="col-5"
>
  <form
    action=""
    class="form-inline"
  >
    <div
      class="form-group mx-sm-3"
    >
      <input
        required=""
        type="text"
        value=""
      >
        
      </input>
    </div>
    <button
      class="btn btn-primary btn-sm"
      type="submit"
    >
      Add
    </button>
  </form>
  <div
    class="row"
  >
    <div
      class="col-10"
    >
      na-na
    </div>
    <div
      class="col"
    >
      <a
        href="#"
      >
        x
      </a>
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="col-10"
    >
      another task
    </div>
    <div
      class="col"
    >
      <a
        href="#"
      >
        x
      </a>
    </div>
  </div>
</div>
`;
